examples/vault-ec2-auth/user-data-vault.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# This script is meant to be run in the User Data of each EC2 Instance while it's booting. The script uses the
+# run-consul script to configure and start Consul in client mode and then the run-vault script to configure and start
+# Vault in server mode. Note that this script assumes it's running in an AMI built from the Packer template in
+# examples/vault-consul-ami/vault-consul.json.
+
+set -e
+
+# Send the log output from this script to user-data.log, syslog, and the console
+# From: https://alestic.com/2010/12/ec2-user-data-output/
+exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+
+# The Packer template puts the TLS certs in these file paths
+readonly VAULT_TLS_CERT_FILE="/opt/vault/tls/vault.crt.pem"
+readonly VAULT_TLS_KEY_FILE="/opt/vault/tls/vault.key.pem"
+
+# The cluster_tag variables below are filled in via Terraform interpolation
+/opt/consul/bin/run-consul --client --cluster-tag-key "${consul_cluster_tag_key}" --cluster-tag-value "${consul_cluster_tag_value}"
+/opt/vault/bin/run-vault --tls-cert-file "$VAULT_TLS_CERT_FILE"  --tls-key-file "$VAULT_TLS_KEY_FILE"
+
+# Initializes a vault server
+# run-vault is running on the background and we have to wait for it to be done,
+# so in case this fails we retry.
+# The boolean operations with the exit status are there to temporarily circumvent the "set -e" at the
+# beginning of this script which exits the script immediatelly for error status.
+for i in $(seq 1 20); do
+  server_output=$(/opt/vault/bin/vault operator init) && exit_status=0 || exit_status=$?
+  if [[ $exit_status -eq 0 ]]; then break; fi
+  sleep 5
+done; (exit $exit_status)
```

Comment:
Same comment here about adding comments (so it's clearer what the script is doing) and putting this in a function with `return` / `exit`.